src/js/actions/search/libraries.js

```diff
@@ -0,0 +1,129 @@
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+define(function (require, exports) {
+    "use strict";
+
+    var Immutable = require("immutable");
+
+    var events = require("js/events"),
+        collection = require("js/util/collection");
+
+    // Map from element IDs to corresponding element
+    var _idMap = {};
+
+    /**
+     * Make list of certain library info so search store can create search options
+     *
+     * @private
+     * @return {Immutable.List.<object>}
+    */
+    var _getLibrarySearchOptions = function () {
+        var libStore = this.flux.store("library"),
+            libraries = libStore.getLibraries();
+
+        _idMap = {};
+
+        return libraries.reduce(function (allMaps, library) {
+            var libMap = Immutable.List();
+            library.elements.forEach(function (element) {
+                var id = element.id.replace(/-/g, "."),
+                    title = element.displayName,
+                    type = element.type,
+                    category = type.substring("application/vnd.adobe.element.".length, type.indexOf("+"));
+
+                category = category.toUpperCase();
+
+                if (category === "CHARACTERSTYLE") {
+                    title = "Character Style Name";
+                }
+                
+                if (category !== "COLOR") {
+                    _idMap[id] = { asset: element, type: category };
+
+                    libMap = libMap.push({
+                        id: id,
+                        name: title,
+                        pathInfo: library.name,
+                        iconID: "tool-rectangle",
+                        category: ["LIBRARY", library.name, category]
+                    });
+                }
+            });
+            return allMaps.concat(libMap);
+        }, Immutable.List());
+    };
+
+    /**
+     * Apply library element when its item is confirmed in search
+     *
+     * @private
+     * @param {string} id ID of layer to select
+    */
+    var _confirmSearch = function (id) {
+        var libActions = this.flux.actions.libraries,
+            elementInfo = _idMap[id];
+
+        if (elementInfo) {
+            var asset = elementInfo.asset;
+            libActions.selectLibrary(asset.library.id);
+
+            switch (elementInfo.type) {
+                case "IMAGE":
+                    var location = { x: 100, y: 100 };
```

Comment:
Instead of using a fixed position, can you use the center of the viewport as the new layer position? You can use uiStore#transformWindowToCanvas to do the transformation.